### PR TITLE
added pagination

### PR DESCRIPTION
**Pull request by piotr-major-rmtcfm-com**

### CODE REVIEW

File: app/controllers/ToDo/IndexController.js [hunk 1]
```diff
@@ -45,8 +45,14 @@ class IndexController {
     }
 
     async invoke(request, response) {
-        const todos = await this.todoRepository.findAll({
-            where: { userId: request.loggedUserId }
+        const { page = 1, perPage = 20 } = request.query;
+
+        const offset = (page - 1) * perPage;
```
Comment (ewelina-szczotka-prpl) [inline app/controllers/ToDo/IndexController.js]: you need to parse `page` and `perPage` to int

File: app/test/controllers/Todo/index.js [hunk 2]
```diff
@@ -38,7 +38,7 @@ describe('Todos', () => {
                 .set('Authorization', 'Bearer ' + loggedUserToken);
 
             for (const todo of todos) {
-                expect(body).to.deep.include({
+                expect(body.rows).to.deep.include({
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/test/controllers/Todo/index.js]: const { body: {rows, count }, status } = response

File: app/controllers/User/IndexController.js [hunk 3]
```diff
@@ -36,7 +36,14 @@ class IndexController {
     }
 
     async invoke(request, response) {
-        const users = await this.userRepository.findAll();
+        const { page = 1, perPage = 20 } = request.query;
+
+        const offset = (page - 1) * perPage;
```
Comment (ewelina-szczotka-prpl) [inline app/controllers/User/IndexController.js]: as above

File: app/test/controllers/Todo/index.js [hunk 4]
```diff
@@ -61,7 +64,9 @@ describe('Todos', () => {
                 });
             }
 
-            expect(statusCode).to.equal(StatusCodes.OK);
+            expect(parseInt(count)).to.deep.equal(todos.length);
+
+            expect(status).to.equal(StatusCodes.OK);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/test/controllers/Todo/index.js]: why now it is status and in previous version it was statusCode?

File: app/test/controllers/Todo/index.js [hunk 5]
```diff
@@ -61,7 +64,9 @@ describe('Todos', () => {
                 });
             }
 
-            expect(statusCode).to.equal(StatusCodes.OK);
+            expect(parseInt(count)).to.deep.equal(todos.length);
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/test/controllers/Todo/index.js]: isnt it Int already?

File: app/test/controllers/Users/index.js [hunk 6]
```diff
@@ -33,7 +33,7 @@ describe('Users', () => {
                 .set('Authorization', 'Bearer ' + loggedUserToken);
 
             for (const user of users) {
-                expect(body).to.deep.include({
+                expect(body.rows).to.deep.include({
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/test/controllers/Users/index.js]: destructure it like above

File: app/validators/paginationValidator.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+const { query } = require('express-validator');
+
+async function isInt(value) {
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/validators/paginationValidator.js]: does returning true/false work?
`custom(page => !isNaN(page)).withMessage('Should be integer!')`?

File: app/validators/paginationValidator.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+const { query } = require('express-validator');
+
+module.exports = [
+    query('page')
+        .optional()
+        .custom(page => !isNaN(parseInt(page)))
```
Comment (daniel-pupczyk-rmtcfm-com) [inline app/validators/paginationValidator.js]: parseInt is not necessary I think

**Review by daniel-pupczyk-rmtcfm-com — APPROVED** [review]

**Pull request merged**